src/flexbox.rs

```diff
@@ -206,31 +203,29 @@ impl Forest {
 
     /// Compute constants that can be reused during the flexbox algorithm.
     #[inline]
-    fn compute_constants(
-        &self,
-        node: NodeId,
-        node_size: Size<Option<f32>>,
-        parent_size: Size<Option<f32>>,
-    ) -> AlgoConstants {
+    fn compute_constants(&self, node: NodeId, node_size: Size<f32>, parent_size: Size<f32>) -> AlgoConstants {
         let dir = self.nodes[node].style.flex_direction;
         let is_row = dir.is_row();
         let is_column = dir.is_column();
         let is_wrap_reverse = self.nodes[node].style.flex_wrap == FlexWrap::WrapReverse;
 
-        let margin = self.nodes[node].style.margin.map(|n| n.resolve(parent_size.width).unwrap_or(0.0));
-        let padding = self.nodes[node].style.padding.map(|n| n.resolve(parent_size.width).unwrap_or(0.0));
-        let border = self.nodes[node].style.border.map(|n| n.resolve(parent_size.width).unwrap_or(0.0));
+        let margin =
+            self.nodes[node].style.margin.map(|n| n.map(|d| d.resolve(parent_size.width)).unwrap_or(Some(0.0)));
+        let padding =
+            self.nodes[node].style.padding.map(|n| n.map(|d| d.resolve(parent_size.width)).unwrap_or(Some(0.0)));
+        let border =
+            self.nodes[node].style.border.map(|n| n.map(|d| d.resolve(parent_size.width)).unwrap_or(Some(0.0)));
 
         let padding_border = Rect {
-            start: padding.start + border.start,
-            end: padding.end + border.end,
-            top: padding.top + border.top,
-            bottom: padding.bottom + border.bottom,
+            start: padding.start.maybe_add(border.start),
+            end: padding.end.maybe_add(border.end),
+            top: padding.top.maybe_add(border.top),
+            bottom: padding.bottom.maybe_add(border.bottom),
         };
 
         let node_inner_size = Size {
-            width: node_size.width.maybe_sub(padding_border.horizontal_axis_sum()),
-            height: node_size.height.maybe_sub(padding_border.vertical_axis_sum()),
+            width: Some(node_size.width.maybe_sub(padding_border.horizontal_axis_sum())),
+            height: Some(node_size.height.maybe_sub(padding_border.vertical_axis_sum())),
```

Comment:
Compiler complains here: How can I satisfy the trait constraints of `MaybeMath` so that these sums work?